modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1466,138 @@ export class Wallet implements IWallet {
+            const sharing = shareOption;
+            assert(sharing.keychain?.toPubKey);
+
+            const secret = getSharedSecret(eckey, Buffer.from(sharing.keychain?.toPubKey, 'hex')).toString('hex');
+            const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: userPrv });
+
+            let pub = keychain.pub ?? keychain.commonPub;
+            if (keychain.commonKeychain) {
+              pub =
+                this.baseCoin.getMPCAlgorithm() === 'eddsa'
+                  ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+                  : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+            }
+
+            sharedKeychain = {
+              pub,
+              encryptedPrv: newEncryptedPrv,
+              fromPubKey: eckey.publicKey.toString('hex'),
+              toPubKey: sharing.keychain?.toPubKey,
+              path: sharing.keychain?.path,
+            };
+          }
+        } catch (e) {
+          if (e.message === 'No encrypted keychains on this wallet.') {
+            sharedKeychain = {};
+          } else {
+            throw e;
+          }
+        }
+        bulkCreateShareOptions.shareOptions.push({
+          user: shareOption.user,
+          permissions: shareOption.permissions,
+          keychain: Object.keys(sharedKeychain ?? {}).length === 0 ? undefined : sharedKeychain,
+        });
+      }
+    }
+    return await this.createBulkKeyShares(bulkCreateShareOptions);
+  }
+
+  /**
+   * Creates bulk key shares for the given wallet.
+   *
+   * @async
+   * @param {BulkCreateShareOptions} [params={ shareOptions: [] }] - The options for creating bulk key shares.
+   * @param {Array<ShareOption>} params.shareOptions - An array of share option objects containing keychain information.
+   * @param {Object} params.shareOptions[].keychain - The keychain object for each share option.
+   * @param {string} params.shareOptions[].keychain.pub - The public key of the keychain.
+   * @param {string} params.shareOptions[].keychain.encryptedPrv - The encrypted private key of the keychain.
+   * @param {string} params.shareOptions[].keychain.fromPubKey - The sender's public key.
+   * @param {string} params.shareOptions[].keychain.toPubKey - The recipient's public key.
+   * @param {string} params.shareOptions[].keychain.path - The derivation path of the keychain.
+   * @throws {Error} If `shareOptions` is empty or any required keychain parameter is missing.
+   * @returns {Promise<any>} A promise that resolves with the result of the bulk key shares creation request.
+   */
+
+  async createBulkKeyShares(
+    params: BulkCreateShareOptions = { shareOptions: [] }
+  ): Promise<CreateBulkWalletShareListResponse> {
+    if (!params.shareOptions || Object.keys(params.shareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    params.shareOptions.forEach((shareOption) => {
+      if (shareOption.keychain) {
```

Comment:
isn't this mandatory in bulk share? We should throw error if keychain is not found

Reply:
not throwing an error/ breaking the call for a missing keychain. However filtering out the shareoptions without keychain/ invalid keychain